File: codechef/LOCMAR16/rectangles.cpp
#include <stdio.h>

const int MAX_SIZE = 505;
int a[MAX_SIZE][MAX_SIZE];
int n, m;

int main()
{
	int t;
	scanf("%d", &t);
	for (int tt = 0; tt < t; tt++)
	{
		scanf("%d %d", &n, &m);
		for (int i = 0; i < n; i++)
			for (int j = 0; j < m; j++)
				scanf("%d", &a[i][j]);

		for (int col = 0; col < m; col++)
		{
			int r = 0, c = col;
			while (r < n && c >= 0) {
				printf("%d ", a[r][c]);
				r++;
				c--;
			}
		}

		for (int row = 1; row < n; row++) 
		{
			int r = row, c = m - 1;
			while (r < n && c >= 0) {
				printf("%d ", a[r][c]);
				r++;
				c--;
			}
		}
		printf("\n");
	}
}

File: codechef/LOCMAR16/rectangles.py
t = int(raw_input())

for i in range(0, t):
	line = raw_input().split()
	n, m = int(line[0]), int(line[1])
	a = []
	for i in range(0, n):
		row = raw_input().split()
		a.append(row)

	for col in range(0, m):
		r, c = 0, col
		while r < n and c >= 0:
			print(a[r][c]),
			r, c = r + 1, c - 1

	for row in range(1, n):
		r, c = row, m - 1
		while r < n and c >= 0:
			print(a[r][c]),
			r, c = r + 1, c - 1